assets/model_monitoring/components/src/model_data_collector_preprocessor/store_url.py

```diff
@@ -190,10 +194,19 @@ def _set_properties(self, ws: Workspace):
             self.path = matches.group("path").strip("/")
             self._datastore = None
         elif url.scheme == "azureml":
-            if ':' in url.path:  # azureml asset path
+            if ':' in url.path:  # azureml:<data_name>:<version> asset path
                 # asset path should be translated to azureml or hdfs path in service, should not reach here
                 raise InvalidInputError("AzureML asset path is not supported as uri_folder.")
-            else:  # azureml long or short form
+
+            data_pattern0 = r"azureml://(subscriptions/([^/]+)/resource[gG]roups/([^/]+)/workspaces/([^/]+)/)?data/(?P<data>[^/]+)/versions/(?P<version>.+)"  # noqa: E501
+            data_pattern1 = r"azureml://locations/([^/]+)/workspaces/([^/]+)/data/(?P<data>[^/]+)/versions/(?P<version>.+)"  # noqa: E501
+            matches = re.match(data_pattern0, self._base_url) or re.match(data_pattern1, self._base_url)
+            if matches:
+                raise InvalidInputError(
+                    f"You are using AzureML data url {self._base_url}, but this format of url is NOT supported as "
+                    "input of Model Monitoring job, please use "
+                    f"azureml:{matches.group('data')}:{matches.group('version')} instead.")
```

Comment:
If the customer specifies this pattern, does it work?

I think the logic here is to verify if it's a V2 dataset

Reply:
As the comment in line 198, my understanding is service(data svc?) will translate data asset url to datastore+path or hdfs path, so although user provide azureml:<data>:<version>, Spark run time got datastore+path or hdfs path(abfss, wasbs).
I think we don't care if it's v1 or v2 dataset? The input is a url, if the url points to a credential v1 dataset, we can also support it?

Reply:
update error msg.